src/components/Settings/Sections/ChatSettings.tsx

```diff
@@ -62,12 +77,18 @@ const ChatSettings = () => {
           <div className="cdx-relative cdx-w-full">
             <input
               required
-              pattern="^(sk(-proj)?-[a-zA-Z0-9]{48}$"
               ref={OpenAiApiKeyInputRef}
               placeholder="sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
               defaultValue={chatSettings.openAIKey || ''}
               type={showPassword ? 'text' : 'password'}
-              className="input"
+              className="input apikey"
+            />
+            <input 
+              ref={OpenAiBaseUrlInputRef}
+              name="openAiBaseUrl" 
+              defaultValue={chatSettings.openAiBaseUrl || 'https://api.openai.com/v1'}
+              placeholder="Enter your OpenAI Base URL" 
+              className="openAiBaseUrl cdx-mt-4 cdx-text-center cdx-p-2 cdx-w-full cdx-rounded-md cdx-border dark:cdx-border-neutral-600 cdx-border-neutral-200 dark:cdx-bg-neutral-800/90 cdx-bg-neutral-200/90 focus:cdx-outline-none focus:cdx-ring-2 focus:cdx-ring-blue-900 focus:cdx-ring-opacity-50 data-[error]:cdx-text-red-500"
             />
```

Comment:
Since there are two inputs now, I think it would be better to use a form element as the parent and assign names to the inputs. This way, inside the onSubmit function, we can handle the data using FormData()

Reply:
Since FieldWrapper is already wrapped as form. I modify the source to get apiKey and baseURL from form element dict. Please review latest source